src/main/java/com/opencart/steps/SearchProductPageBL.java

```diff
@@ -25,26 +26,60 @@ public SearchProductPageBL makeSearch(String value) {
         searchProductPage.getInputSearchCriteria().clear();
         searchProductPage.getInputSearchCriteria().sendKeys(value);
         clickOnSearchGroupButton();
+        clickOnSelectButton();
+        clickOnSearchButton();
         searchProductResultPage = new SearchProductResultPage();
         return this;
     }
 
-    public void verifySearchResult(String searchCriteria) {
-        Assert.assertTrue(findAllProductBySearchCriteria(searchCriteria));
+    public void verifyPositiveSearchResult(String searchCriteria) {
+        SoftAssert softAssert = new SoftAssert();
+        String expectedResult = "Products meeting the search criteria";
+        String actualResult = searchProductResultPage.getSuccessTitle().getText();
+        softAssert.assertEquals(expectedResult, actualResult, "Title is correct, now you can check the list of products");
+        softAssert.assertTrue(findAllProductBySearchCriteria(searchCriteria),
+                "Check the list of products Done!");
+        softAssert.assertAll();
+    }
+
+    public void verifyNegativeSearchResult() {
+        SoftAssert softAssert = new SoftAssert();
+        String expectedResult = "There is no product that matches the search criteria.";
+        String actualResult = searchProductResultPage.getNoProductsFoundTitle().getText();
+        softAssert.assertEquals(expectedResult, actualResult, "Search criteria is actually correct");
+        softAssert.assertAll();
     }
 
     private boolean findAllProductBySearchCriteria(String searchCriteria) {
         List<ProductContainer> productContainers = searchProductResultPage.getProductContainers();
+        boolean isProductListContainsSearchCriteria = true;
         for (ProductContainer productContainer : productContainers) {
             if (!productContainer.getName().contains(searchCriteria)) {
-                return false;
+                isProductListContainsSearchCriteria = false;
             }
         }
-        return true;
+        return isProductListContainsSearchCriteria;
     }
 
     private SearchProductPageBL clickOnSearchGroupButton() {
         searchProductPage.getInputSearchGroupButton().click();
         return this;
     }
+
+    private SearchProductPageBL clickOnSearchButton() {
+        searchProductPage.getSearchButton().click();
+        return this;
+    }
+
+    private SearchProductPageBL clickOnSelectButton() {
+        Select select = new Select(searchProductPage.getCategoryContainers());
+        select.selectByValue("20");
+
+//        Think how to make it more flexible?
+//        List<WebElement> optionList = select.getOptions();
+//        for (WebElement option : optionList){
+//            option.getCssValue("20");
+//        }
```

Comment:
You can just put the name of the select option as a method parameter, and then find the appropriate name within dropdown select options and click on it. 
Also, need to rename this method on - "selectCategoryFromDropdown" or something like that